### PR TITLE
ci: runs TinyGo stdlib tests on mac and windows

### LINKED ISSUE #1039: Pass TinyGo wasi tests
TinyGo has a target `make test-tinygo-wasi` which ends up running the following via wasmtime. We should be able to do the same via wazero

```bash
$ tinygo test -target wasi -c compress/zlib container/heap container/list container/ring crypto/des crypto/internal/subtle crypto/md5 crypto/rc4 crypto/sha1 crypto/sha256 crypto/sha512 debug/macho embed/internal/embedtest encoding encoding/ascii85 encoding/base32 encoding/csv encoding/hex go/scanner hash hash/adler32 hash/crc64 hash/fnv html internal/itoa internal/profile math math/cmplx net net/http/internal/ascii net/mail os path reflect sync testing testing/iotest text/scanner unicode unicode/utf16 unicode/utf8  crypto/internal/nistec/fiat compress/bzip2 crypto/dsa index/suffixarray  ./tests/runtime_wasi
```

### PR DESCRIPTION
**Pull request by mathetake**
Closes #1039 

Signed-off-by: Takeshi Yoneda <takeshi@tetrate.io>

### CODE REVIEW

**Comment by mathetake** [comment]
lol https://github.com/golang/go/blob/40a0986959167e7b82ae16a35a5dd2a347609eff/src/compress/zlib/writer_test.go#L16-L30

**Comment by mathetake** [comment]
cool

File: .github/workflows/integration.yaml [hunk 1]
```diff
@@ -207,6 +192,54 @@ jobs:
...
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Install wazero
+        run: go install ./cmd/wazero
+
+      - name: Run standard library tests
+        run: |
+          for bin in *.test; do
```
Comment (mathetake) [inline .github/workflows/integration.yaml:239]: it turns out that the majority of the time was spent on compiling wasm binary by tinygo, so it seems all test finishes in like 30 sec or so.

File: .github/workflows/integration.yaml [hunk 2]
```diff
@@ -177,7 +163,6 @@ jobs:
             crypto/sha1 \
             crypto/sha256 \
             crypto/sha512 \
-            debug/macho \
```
Comment (codefromthecrypt) [inline .github/workflows/integration.yaml:180]: no macho no mas

File: .github/workflows/integration.yaml (hunk 1 above)
Comment (codefromthecrypt) [reply .github/workflows/integration.yaml:239]: good news cc @dgryski

**Review by codefromthecrypt — APPROVED** [review]
cool I'm glad to see you re-using the "build if not distributed" approach to ensure we can test all platforms wazero's compiler engine should work on!

**Pull request merged**